### ISSUE TITLE
Torch backend does not support "same" padding in average pooling

### ISSUE THREAD

**Issue by chenmoneygithub** [keras-team/keras#18452]
Torch does not support uneven padding in its `AvgPool` ops, which causes our average pooling layers to behave different in Torch backend from other backends. Our tmp fix is to pad evenly using the shorter-side padding value so that most of output values are the same as `tf.keras`, except from the last few colums/rows. 

We cannot bypass the issue by padding ahead before pooling, because we need to ignore padding value during pooling, but Torch does not provide a way to exclude those padded values.

**Comment by sachinprasadhs** [comment]
The feature has been implemented in the above linked PR and the commit https://github.com/keras-team/keras-core/pull/372/commits/e438a401472bd523736addecafb425d54dc7209e.
Can we close this issue?
